command/cp.go

```diff
@@ -956,3 +998,57 @@ func guessContentType(file *os.File) string {
 	}
 	return contentType
 }
+
+type CustomWriter struct {
+	c  Copy
+	fp *os.File
+}
+
+func (r *CustomWriter) WriteAt(p []byte, off int64) (int, error) {
+	n, err := r.fp.Write(p)
+	if err != nil {
+		return n, err
+	}
+
+	r.c.progressbar.AddCompletedBytes(n)
+
+	return n, err
+}
+
+type CustomReader struct {
```

Comment:
Same comments as the Writer applies to the Reader as well.

Reply:
Resolved. https://github.com/peak/s5cmd/pull/590/commits/cdcaf16860b80f1531caf4d87207e3e7f22d44c1